addons/editor-theme3/theme3.js

```diff
@@ -615,6 +615,35 @@ export default async function ({ addon, console, msg }) {
     Object.assign(originalColors, newColors);
   };
 
+  (async () => {
+    // Custom colors for "Add an input/label" block icons in the "Make a block" popup menu, by pumpkinhasapatch
+    const uriHeader = "data:image/svg+xml;base64,";
+    const myBlocksCategory = categories[9];
+    while (true) {
+      // Wait until "Make a block" popup is opened and icon elements are created
+      const iconElement = await addon.tab.waitForElement("[class^=custom-procedures_option-icon_]", {
+        markAsSeen: true,
```

Comment:
```suggestion
        markAsSeen: true,
        reduxEvents: ["scratch-gui/custom-procedures/ACTIVATE_CUSTOM_PROCEDURES"],
        reduxCondition: (state) => state.scratchGui.editorTab.activeTabIndex === 0 && !state.scratchGui.mode.isPlayerOnly,
```
These options allow `waitForElement` to apply optimizations.

Reply:
Thanks for the feedback! I have not learned about redux, but I guess this sort of unloads the addon and stops checking for new icon elements when on the project page or fullscreen mode where it is not possible to open the "Make a block" modal.

Reply:
Can we further optimize this code, so it only checks if there are icons when a modal popup or the "Make a blocks" modal is opened instead of running a waitForElement all the time in the project editor?

Reply:
> Can we further optimize this code, so it only checks if there are icons when a modal popup or the "Make a blocks" modal is created? It would be better than running a waitForElement all the time in the project editor.

The `reduxEvents` option does exactly that! And `reduxCondition` does what you guessed - it tells the `waitForElement` API to only check if the icons exist when inside the code editor.

Reply:
Unfortunately the only way to find what these options should be set to look at the Scratch source code (or copy them from other addons that change the same part of the editor).